### PR TITLE
fix: don't fail with overwrite

### LINKED ISSUE #31: A successful overwrite should not fail the run
#30 broke this

### PR DESCRIPTION
**Pull request by figsoda**
fixes #31

### CODE REVIEW

File: src/runner.rs [hunk 1]
```diff
@@ -77,7 +77,7 @@ impl Runner {
                 })
                 .collect();
 
-            if !failures.is_empty() {
+            if !failures.is_empty() && mode != &Mode::Overwrite {
```
Comment (epage) [inline src/runner.rs:80]: Is there any case where we can still fail with overwrite?

#189 takes care of one.  

A safer approach might be to swallow the errors earlier, where we know what the overwrite did

**Comment by figsoda** [comment]
I will work on it after #189 is out of the way so I don't get merge conflicts

**Comment by epage** [comment]
@figsoda thinking about this more, this is an issue when the user is directly interacting with the program.  The current behavior is a bug.  What this PR introduces also has bugs.  The question is which bug is worse.  I think I lean towards the current behavior being worse.  What do you think?

If we agree the current behavior is worse, we could possibly merge this now, open a new issue, and then work to get that new issue fixed.  This would make it so people have the better experience sooner.  Thoughts?

**Comment by figsoda** [comment]
I agree, the current behavior basically fails every single time on TRYCMD=overwrite and doesn't give much information as an interactive command

**Pull request merged**